game/controllers-midi.cc

```diff
@@ -37,25 +37,25 @@ namespace input {
 		}
 		bool process(Event& event) override {
 			PmEvent ev;
-			for (auto it = m_streams.begin(); it != m_streams.end(); ++it) {
-				if (Pm_Read(*it->second, &ev, 1) != 1) continue;
-				unsigned char evnt = ev.message & 0xF0;
-				unsigned char note = ev.message >> 8;
-				unsigned char vel  = ev.message >> 16;
-				unsigned chan = (ev.message & 0x0F) + 1;  // It is conventional to use one-based indexing
-				if (evnt == 0x80 /* NOTE OFF */) { evnt = 0x90; vel = 0; }  // Translate NOTE OFF into NOTE ON with zero-velocity
+			for (auto const& [key, value]: m_streams) { 
+				if (Pm_Read(value.get(), &ev, 1) != 1) continue;
+				unsigned char evnt = static_cast<unsigned char>(ev.message & 0xF0);
+				unsigned char note = static_cast<unsigned char>(ev.message >> 8);
+				unsigned char vel  = static_cast<unsigned char>(ev.message >> 16);
```

Comment:
what about using http://portmedia.sourceforge.net/portmidi/doxygen/group__grp__events__filters.html#g5e4ff2b1f4d8ae84ddeaa6465d5de5db macros?

Reply:
I'm not really sure which ones, tbh.

Reply:
done